test/inductor/test_triton_kernels.py

```diff
@@ -3550,6 +3550,155 @@ def f(x, increment):
 
         self.assertEqual(y + increment, x)
 
+    """
+    The behavior of prune_configs_by deviates from OSS Triton in that user-defined functions are
+    executed in Dynamo instead of the standard CPython interpreter.
+
+    An example of how user-defined Triton kernels differ from OSS Triton:
+
+    records = {}
+    def early_config_prune(configs, named_args, **kwargs):
+        records["run_early_config_prune"] = True
+        if "N" in kwargs and kwargs["N"] == 1024:
+            records["capture_kwargs"] = True
+        elif "N" in kwargs:
+            breakpoint()
+        if "dst" in named_args and "src" in named_args and len(named_args) == 5:
+            records["capture_named_args"] = True
+        return [configs[0]]
+
+    In OSS Triton this user-defined function executes correctly in the test.
+    We execute early_config_prune in dynamo, so `records` is not updated.
```

Comment:
As discussed offline with @zou3519, Dynamo should support updating the nonlocals / globals, so we should try to fix this.